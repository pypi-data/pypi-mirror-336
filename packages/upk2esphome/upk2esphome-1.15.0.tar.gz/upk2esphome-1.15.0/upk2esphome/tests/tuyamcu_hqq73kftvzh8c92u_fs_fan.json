{
	"category": "fs",
	"model": {
		"modelId": "0000003niz",
		"services": [
			{
				"actions": [],
				"code": "",
				"description": "",
				"events": [],
				"name": "默认服务",
				"properties": [
					{
						"abilityId": 1,
						"accessMode": "rw",
						"code": "switch",
						"description": "【必选】",
						"extensions": {
							"iconName": "icon-power"
						},
						"name": "开关",
						"typeSpec": {
							"type": "bool",
							"typeDefaultValue": false
						}
					},
					{
						"abilityId": 3,
						"accessMode": "rw",
						"code": "fan_speed",
						"description": "【必选】可根据实际档位需求扩展",
						"extensions": {
							"iconName": "icon-a_fan_low"
						},
						"name": "风速",
						"typeSpec": {
							"range": [
								"1",
								"2",
								"3"
							],
							"type": "enum",
							"typeDefaultValue": "1"
						}
					},
					{
						"abilityId": 6,
						"accessMode": "rw",
						"code": "countdown",
						"description": "0：取消\n1：1小时\n2：2小时\n.....",
						"extensions": {
							"iconName": "icon-dp_time2"
						},
						"name": "倒计时",
						"typeSpec": {
							"range": [
								"0",
								"1",
								"2",
								"3",
								"4",
								"5",
								"6"
							],
							"type": "enum",
							"typeDefaultValue": "0"
						}
					},
					{
						"abilityId": 7,
						"accessMode": "ro",
						"code": "countdown_left",
						"description": "",
						"extensions": {
							"iconName": "icon-dp_time3"
						},
						"name": "倒计时剩余时间",
						"typeSpec": {
							"max": 360,
							"min": 0,
							"scale": 0,
							"step": 1,
							"type": "value",
							"typeDefaultValue": 0,
							"unit": "分钟"
						}
					},
					{
						"abilityId": 9,
						"accessMode": "rw",
						"code": "light",
						"description": "",
						"extensions": {
							"iconName": "icon-tcl_function_light",
							"attribute": "16"
						},
						"name": "灯光",
						"typeSpec": {
							"type": "bool",
							"typeDefaultValue": false
						}
					}
				]
			}
		]
	}
}
