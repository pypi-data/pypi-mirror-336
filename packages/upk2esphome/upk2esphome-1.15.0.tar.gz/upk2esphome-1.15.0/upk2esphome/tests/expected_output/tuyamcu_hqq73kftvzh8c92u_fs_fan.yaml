uart:
  rx_pin: RX1
  tx_pin: TX1
  baud_rate: 115200

tuya:
  # DPIDs processed from schema model: 0000003niz

switch:
  - platform: tuya
    switch_datapoint: 1
    name: (Unconfirmed) Switch
  - platform: tuya
    switch_datapoint: 9
    name: (Unconfirmed) Light

select:
  - platform: tuya
    enum_datapoint: 3
    name: (Unconfirmed) Fan Speed
    optimistic: true
    options:
      0: "1"
      1: "2"
      2: "3"
  - platform: tuya
    enum_datapoint: 6
    name: (Unconfirmed) Countdown
    optimistic: true
    options:
      0: "0"
      1: "1"
      2: "2"
      3: "3"
      4: "4"
      5: "5"
      6: "6"

sensor:
  - platform: tuya
    sensor_datapoint: 7
    name: (Unconfirmed) Countdown Left
    unit_of_measurement: ""
